src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -371,12 +449,55 @@ public override async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseA
                     }
                 }
 
-                yield return update;
+                if (functionCallContents?.Count is not > 0 || !hasApprovalRequiringFunctions)
+                {
+                    // If there are no function calls to make yet, or if none of the functions require approval at all,
+                    // we can yield the update as-is.
+                    lastYieldedUpdateIndex++;
+                    yield return update;
+                    Activity.Current = activity; // workaround for https://github.com/dotnet/runtime/issues/47802
+                }
+                else
+                {
+                    // Check if any of the function call contents in this update requires approval.
+                    // Once we find the first one that requires approval, this method becomes a no-op.
+                    (hasApprovalRequiringFcc, lastApprovalCheckedFCCIndex) = await CheckForApprovalRequiringFCCAsync(
+                        functionCallContents, approvalRequiredFunctions, hasApprovalRequiringFcc, lastApprovalCheckedFCCIndex, cancellationToken);
+
+                    // We've encountered a function call content that requires approval (either in this update or ealier)
```

Comment:
Consider turning on the spell checker in VS, there's multiple typos in this file.
```suggestion
                    // We've encountered a function call content that requires approval (either in this update or earlier)
```